SimDataFormats/CaloAnalysis/interface/MtdSimLayerCluster.h

```diff
@@ -0,0 +1,300 @@
+  math::XYZVectorF momentum() const { return p4().Vect(); }
+
+  /// @brief Vector to boost to the particle centre of mass frame.
+  math::XYZVectorF boostToCM() const { return p4().BoostToCM(); }
+
+  /// @brief Magnitude of momentum vector. Note this is taken from the first
+  /// SimTrack only.
+  float p() const { return p4().P(); }
+
+  /// @brief Energy. Note this is taken from the first SimTrack only.
+  float energy() const { return p4().E(); }
+
+  /// @brief Transverse energy. Note this is taken from the first SimTrack only.
+  float et() const { return p4().Et(); }
+
+  /// @brief Mass. Note this is taken from the first SimTrack only.
+  float mass() const { return p4().M(); }
+
+  /// @brief Mass squared. Note this is taken from the first SimTrack only.
+  float massSqr() const { return pow(mass(), 2); }
+
+  /// @brief Transverse mass. Note this is taken from the first SimTrack only.
+  float mt() const { return p4().Mt(); }
+
+  /// @brief Transverse mass squared. Note this is taken from the first SimTrack
+  /// only.
+  float mtSqr() const { return p4().Mt2(); }
+
+  /// @brief x coordinate of momentum vector. Note this is taken from the first
+  /// SimTrack only.
+  float px() const { return p4().Px(); }
+
+  /// @brief y coordinate of momentum vector. Note this is taken from the first
+  /// SimTrack only.
+  float py() const { return p4().Py(); }
+
+  /// @brief z coordinate of momentum vector. Note this is taken from the first
+  /// SimTrack only.
+  float pz() const { return p4().Pz(); }
+
+  /// @brief Transverse momentum. Note this is taken from the first SimTrack
+  /// only.
+  float pt() const { return p4().Pt(); }
+
+  /// @brief Momentum azimuthal angle. Note this is taken from the first
+  /// SimTrack only.
+  float phi() const { return p4().Phi(); }
+
+  /// @brief Momentum polar angle. Note this is taken from the first SimTrack
+  /// only.
+  float theta() const { return p4().Theta(); }
+
+  /// @brief Momentum pseudorapidity. Note this is taken from the simtrack
+  /// before the calorimeter
+  float eta() const { return p4().Eta(); }
+
+  /// @brief Rapidity. Note this is taken from the simtrack before the
+  /// calorimeter
+  float rapidity() const { return p4().Rapidity(); }
+
+  /// @brief Same as rapidity().
+  float y() const { return rapidity(); }
+
+  /** @brief Status word.
+   *
+   * Returns status() from the first gen particle, or -99 if there are no gen
+   * particles attached. */
+  int status() const { return genParticles_.empty() ? -99 : (*genParticles_[0]).status(); }
+
+  static const unsigned int longLivedTag;  ///< long lived flag
+
+  /// is long lived?
+  bool longLived() const { return status() & longLivedTag; }
+
+  /** @brief Gives the total number of SimHits, in the cluster */
+  int numberOfSimHits() const { return nsimhits_; }
+
+  /** @brief Gives the total number of SimHits, in the cluster */
+  int numberOfRecHits() const { return hits_.size(); }
+
```

Comment:
@AuroraPerego this name is misleading, isn't it? I understand there is no reco information in this object, and these two methods above should provide the same result, is it correct?